src/Ui/RestGetTicket.php

```diff
@@ -25,32 +31,76 @@
 use Horde_Session;
 use Horde\Log\Logger;
 
-
-
-
 /**
  * Controller class for Public UI
  */
 class RestGetTicket extends RestBase
 {
-    
-    /**
-     * Overload this method for actually implementing stuff
-     * 
-     * Quick & Dirty: Use $this->injector to get application services
-     * Proper: Overload and amend constructor
-     */
+    protected function getResponseStream(array $data)
+    {
+        return $this->streamFactory->createStream(json_encode($data));
+    }
+
     protected function buildResponseStream(ServerRequestInterface $request): ?StreamInterface
     {
-        $content = json_encode([
-            'ticket' => [
-                'ticket_id' => 'XYZ0XYZ1XYZ0XYZ1XYZ0XYZ1XYZ0XYZ1',
-                'ticket_time' => (string) new \Horde_Date('2021-11-24 10:30', 'UTC')
-            ]
-        ]);
-        if ($content) {
-            return $this->streamFactory->createStream($content);
+        $method = $request->getMethod();
+        if ($method !== "POST") {
+            return $this->getResponseStream(
+                ['error' => 'request method needs to be post']
+            );
+        }
+        $body = $request->getBody()->getContents();
+        $data = json_decode($body, true);
```

Comment:
missing type check on $data - returns NULL for invalid json

Reply:
Alternatively, use JSON_THROW_ON_ERROR  to get exceptions rather than false. https://www.php.net/manual/en/class.jsonexception.php